ab2d-aggregator/src/main/java/gov/cms/ab2d/aggregator/BeneficiaryStream.java

```diff
@@ -0,0 +1,90 @@
+package gov.cms.ab2d.aggregator;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+/**
+ * This manages all the creating and streaming of beneficiary data. To use this, you create a try
+ * with resources with the BeneficiaryStream. For example:
+ *
+ * try (BeneficiaryStream stream = new BeneficiaryStream(jobId, false)) {
+ *     for (int i = 0; i < beneBatchSize; i++) {
+ *         stream.write(getNdJson(benes.get(i)));
+ *     }
+ * } catch(Exception ex) { }
+ *
+ * This does all the work of creating a temporary file in the correct location to stream, allowing
+ * streams to be written to and when the file is closed, it moves it to the "done" directory, waiting
+ * for the aggregator to pick it up
+ */
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class BeneficiaryStream implements AutoCloseable {
+    private static final String FILE_PREFIX = "tmp_";
+    private final BufferedOutputStream bout;
+    private final File tmpFile;
+    private final File completeFile;
+    private final FileOutputType type;
+    private final String jobDir;
+    private boolean open;
+    private final FileOutputStream stream;
+    private final String streamingDir;
+
+    public BeneficiaryStream(String jobId, String baseDir, FileOutputType type, String streamingDir, String finishedDir) throws IOException {
+        this.type = type;
+        this.open = true;
+        this.jobDir = Path.of(baseDir, jobId).toFile().getAbsolutePath();
+        this.streamingDir = streamingDir;
+        JobHelper.workerSetUpJobDirectories(jobId, baseDir, streamingDir, finishedDir);
+        this.tmpFile = createNewFile();
+        this.stream = new FileOutputStream(tmpFile);
+        this.bout = new BufferedOutputStream(stream);
```

Comment:
I'm not sure what it is and I'm not sure how big the temporary files will be. That will depend on how many beneficiaries are in each batch and how many eobs each has. There will be balancing act between number of threads, number of benes per file, number of files created, and how effectively the aggregator can combine files (if they're too big it will be hard to guarantee they are under the max size and/or the combined size is not close to the max size). Either way, it should probably be a variable we can tweak.